drivers/nutdrv_qx_masterguard.c

```diff
@@ -0,0 +1,1049 @@
+	 *    012345678901234567
+	 *    0         1
+	 *    (00,1869:19:06:37
+	 */
+	/* type			flags		rw	command		answer	len	leading	value	from	to	dfl	qxflags			precmd				preans	preproc */
+	/* this is not really the uptime, but the running time since last maintenance */
+	{ "device.uptime",	ST_FLAG_RW,	NULL,	"DRC,XX\r",	"",	17,	'(',	"",	4,	16,	"%.0f",	QX_FLAG_SEMI_STATIC,	masterguard_add_slaveaddr,	NULL,	masterguard_tttt_hh_mm_ss },
+	{ "device.uptime",	0,		NULL,	"SRC%s,XX\r",	"",	0,	'\0',	"",	0,	0,	"thms",	QX_FLAG_SETVAR,		masterguard_add_slaveaddr,	NULL,	masterguard_setvar },
+
+	/*
+	 * > [MSO\r]
+	 * < [(220 230 240\r]
+	 *    0123456789012
+	 *    0         1
+	 *    (220 230 240
+	 */
+	/* type			flags	rw	command		answer	len	leading	value	from	to	dfl	qxflags				precmd	preans	preproc */
+	{ "output_voltages",	0,	NULL,	"MSO\r",	"",	5,	'(',	"",	1,	0,	"%s",	QX_FLAG_STATIC | QX_FLAG_NONUT,	NULL,	NULL,	masterguard_output_voltages },
+
+	/*
+	 * > [PNV\r]
+	 * < [(PNV=nnn\r]
+	 *    012345678
+	 *    0
+	 *    (PNV=230
+	 */
+	/* type				flags		rw			command		answer	len	leading	value	from	to	dfl	qxflags					precmd	preans	preproc */
+	{ "output.voltage.nominal",	ST_FLAG_RW,	NULL /* see claim */,	"PNV\r",	"",	8,	'(',	"",	5,	7,	"%.0f",	QX_FLAG_SEMI_STATIC | QX_FLAG_ENUM,	NULL,	NULL,	NULL },
+	{ "output.voltage.nominal",	0,		NULL,			"PNV=%03d\r",	"",	0,	'\0',	"",	0,	0,	"d",	QX_FLAG_SETVAR,				NULL,	NULL,	masterguard_setvar },
+	{ "output.current.nominal",	0,		NULL,			"PNV\r",	"",	8,	'(',	"",	5,	7,	"%.0f",	QX_FLAG_SEMI_STATIC,			NULL,	NULL,	masterguard_output_current_nominal },
+
+	/*
+	 * > [FLT,XX\r]
+	 * < [(XX,A aaaa TTTT:hh:mm:ss B bbbb TTTT:hh:mm:ss C cccc TTTT:hh:mm:ss D dddd TTTT:hh:mm:ss E eeee TTTT:hh:mm:ss\r
+	 *    0123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678
+	 *    0         1         2         3         4         5         6         7         8         9         0
+	 *    (00,7 0043 0000:16:48:06 0 0000 0000:00:00:00 0 0000 0000:00:00:00 0 0000 0000:00:00:00 0 0000 0000:00:00:00
+	 *    (01,9 0010 1780:14:57:19 7 0046 0000:21:14:41 0 0000 0000:00:00:00 0 0000 0000:00:00:00 0 0000 0000:00:00:00
+	 */
+	/* type		flags	rw	command		answer	len	leading	value	from	to	dfl	qxflags		precmd				preans	preproc */
+	{ "fault_1",	0,	NULL,	"FLT,XX\r",	"",	108,	'(',	"",	4,	23,	"%s",	QX_FLAG_NONUT,	masterguard_add_slaveaddr,	NULL,	masterguard_fault },
+	{ "fault_2",	0,	NULL,	"FLT,XX\r",	"",	108,	'(',	"",	25,	44,	"%s",	QX_FLAG_NONUT,	masterguard_add_slaveaddr,	NULL,	masterguard_fault },
+	{ "fault_3",	0,	NULL,	"FLT,XX\r",	"",	108,	'(',	"",	46,	65,	"%s",	QX_FLAG_NONUT,	masterguard_add_slaveaddr,	NULL,	masterguard_fault },
+	{ "fault_4",	0,	NULL,	"FLT,XX\r",	"",	108,	'(',	"",	67,	86,	"%s",	QX_FLAG_NONUT,	masterguard_add_slaveaddr,	NULL,	masterguard_fault },
+	{ "fault_5",	0,	NULL,	"FLT,XX\r",	"",	108,	'(',	"",	88,	107,	"%s",	QX_FLAG_NONUT,	masterguard_add_slaveaddr,	NULL,	masterguard_fault },
```

Comment:
@aquette @clepple : Any suggestions for standard field naming?

Reply:
Maybe `ups.fault.N` could be a more detailed version of `ups.alarm` (which is just space-separated keywords)?